api/src/wfl/module/covid.clj

```diff
@@ -157,34 +195,105 @@
           row-ids (find-new-rows source now)]
       (when-not (empty? row-ids)
         (write-snapshots source (make-snapshots source row-ids)))
       (update-last-checked source now))))
 
-(defn ^:private load-tdr-source [{:keys [source-ptr] :as workload}]
-  {:type    tdr-source-type
-   :queue   nil
-   :updated nil})
+(defn ^:private load-tdr-source [tx {:keys [source_items] :as details}]
+  (if-let [source (->> (util/parse-int source_items)
+                       (conj ["SELECT * FROM TerraDataRepoSource WHERE id = ? LIMIT 1"])
+                       (jdbc/query tx)
+                       first)]
+    (assoc source :type tdr-source-type)
+    (throw (ex-info (str "No source matching id " source_items) details))))
 
 ;; maybe we want to split check-request and create to avoid writing
 ;; records for bad workload requests?
-(defn ^:private create-tdr-source [source-request]
-  (letfn [(check-request! [request] (throw (Exception. "Not implemented")))
-          (make-record    [request] nil)]
-    (check-request! source-request)
-    (let [foreign-key (make-record source-request)]
-      (load-tdr-source {:source-ptr foreign-key}))))
-
-(defoverload workloads/start-workload! pipeline start-covid-workload!)
-
-(defoverload create-source! tdr-source-type create-tdr-source)
-(defoverload peek-queue!    tdr-source-type peek-tdr-source-queue)
-(defoverload pop-queue!     tdr-source-type pop-tdr-source-queue)
+(defn ^:private create-tdr-source [tx id request]
+  (let [create    "CREATE TABLE %s OF TerraDataRepoSourceDetails (PRIMARY KEY (id))"
+        details   (format "%sDetails_%09d" tdr-source-type id)
+        _         (jdbc/execute! tx [(format create details)])
+        req->cols {:dataset :dataset
+                   :table   :dataset_table
+                   :column  :table_column_name}
+        items     (-> (select-keys request (keys req->cols))
+                      (set/rename-keys req->cols)
+                      (assoc :details details)
+                      (->> (jdbc/insert! tx tdr-source-type)))]
+    [tdr-source-type (-> items first :id)]))
+
+(defoverload create-source! tdr-source-name create-tdr-source)
+(defoverload peek-queue! tdr-source-type peek-tdr-source-queue)
+(defoverload pop-queue! tdr-source-type pop-tdr-source-queue)
 (defoverload update-source! tdr-source-type update-tdr-source)
-(defoverload load-source!   tdr-source-type load-tdr-source)
+(defoverload load-source! tdr-source-type load-tdr-source)
+
+(def ^:private terra-executor-name "Terra")
+(def ^:private terra-executor-type "TerraExecutor")
+
+(defn ^:private create-terra-executor [tx id request]
+  (let [create    "CREATE TABLE %s OF TerraExecutorDetails (PRIMARY KEY (id))"
+        details   (format "%sDetails_%09d" terra-executor-type id)
+        _         (jdbc/execute! tx [(format create details)])
+        req->cols {:workspace                  :workspace
+                   :methodConfiguration        :method_configuration
```

Comment:
Is there any reason we don't use hyphens in our database? In either case, I think we should return to the user what we receive so decide on one naming format. I prefer kebabs and camels to snakes. Pick one.

Reply:
Sticking with camelCase